AsyncDisplayKit/ASTableView.mm

```diff
@@ -1454,6 +1463,33 @@ - (void)rangeController:(ASRangeController *)rangeController didEndUpdatesAnimat
   }
 }
 
+- (void)rangeControllerDidReloadData:(ASRangeController *)rangeController
+{
+  ASDisplayNodeAssertMainThread();
+  LOG(@"UITableView reloadData");
+  
+  if (!self.asyncDataSource) {
+    return; // if the asyncDataSource has become invalid while we are processing, ignore this request to avoid crashes
+  }
+  
+  ASPerformBlockWithoutAnimation(YES, ^{
+    if (self.test_enableSuperUpdateCallLogging) {
+      NSLog(@"-[super reloadData]");
+    }
+    [super reloadData];
+    if (!_performingBatchUpdates) {
+      [_rangeController updateIfNeeded];
+    }
+    // TODO Should reloadData triggers batch fetching?
+//    [self _scheduleCheckForBatchFetchingForNumberOfChanges:];
+  });
+  
+  //TODO is _automaticallyAdjustsContentOffset needed for reloadData?
+//  if (_automaticallyAdjustsContentOffset) {
+//    [self adjustContentOffsetWithNodes:nodes atIndexPaths:indexPaths inserting:YES];
```

Comment:
Hmm good question. I think it should not apply to `reloadData` because we have to assume that every single item was replaced by something completely different.